src/cross_section/include/cross_section.h

```diff
@@ -54,6 +56,8 @@ class CrossSection {
   CrossSection Translate(glm::vec2 v);
   CrossSection Rotate(float degrees);
   CrossSection Scale(glm::vec2 s);
+  CrossSection Mirror(glm::vec2 ax, bool reverse = true);
+  CrossSection Rewind();
```

Comment:
Is this necessary?

Reply:
I thought it would be helpful to include when I was testing Mirror, before I added the reverse parameter. Mirroring reverses the paths, so outlines are now holes and vice-versa. This is something that doesn't come up in OpenSCAD, as I imagine they counter the reversal of the paths for you. I figured it's conceivable that someone might feed contours into CrossSection and find that they are the wrong winding, so having Rewind provides an easy way to remedy that without having to mess with iterators or loops to fix their input vectors.

To bleed across reviews, besides mirroring on arbitrary vectors/axes, this is another reason why I think Mirror is better than negative scaling. To be honest, I think I'd remove the reverse parameter since I'm not sure people would expect or want their cross section to invert/dissapear on mirroring.

Reply:
Yeah, I hear you, but my main design goal is to mimic the `Manifold` API, and we don't have an invert function. If it becomes important, we can add it to both at once down the road.

Reply:
I’m wondering if we should automatically check for winding direction and help users fix their cross section before creating a solid from it.

Reply:
This is equivalent to a having a manifold with the triangles wound CW instead of CCW. I wouldn't want users to accidentally use the wrong winding and have us secretly fixing it as best we can. I'm open to throwing an exception if they're enabled when the output is empty. 

Alternative: we had mentioned earlier the possibility of specifying an optional fill rule in the constructor - non-zero would take care of this. Then at least it's explicit.

Reply:
Since contours can be an arbitrary number of polygons including nesting and overlaps, I don't think the intended winding is easily knowable. If the input is a single contour, or a data type that is understood to only represent a single outline with zero or more holes, then doing that could make sense (maybe with a usually on optional parameter in case they intended it for some reason, like to do clipping with the opposite winding polygon with another to achieve with inverted mean ii ng or something 😂.

Reply:
> I'm open to throwing an exception if they're enabled when the output is empty.

As long as it can be turned off since exceptions aren't great for FFI.

> Alternative: we had mentioned earlier the possibility of specifying an optional fill rule in the constructor - non-zero would take care of this. Then at least it's explicit.

Unfortunately carrying around a fillrule from construction time wouldn't really work, since there would be CrossSections with mismatching fillrules getting out into booleans with eachother. It would need to be an optional on every method/constructor that does a boolean.

Reply:
No, the specified fill rule would only affect the initial cleanup on construction. That guarantees every CrossSection has no overlaps, so the fill rule no longer matters for any operations thereafter.

Reply:
Oh gotcha 👍.

Reply:
I think I still had my wires crossed for a bit with the lazy cleaning mindset (which is *sort of* like what I have in my other Clipper bindings). This is definitely a win for the clean on construction scheme.

Reply:
Rewind is removed, and FillRule has been added as an optional (default Positive) to the contour constructors.